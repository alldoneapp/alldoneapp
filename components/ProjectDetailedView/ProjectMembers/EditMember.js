import React, { useEffect, useRef, useState } from 'react'
import { Keyboard, StyleSheet, View } from 'react-native'
import Button from '../../UIControls/Button'
import Icon from '../../Icon'
import store from '../../../redux/store'
import styles, { colors } from '../../styles/global'
import Backend from '../../../utils/BackendBridge'
import HelperFunctions from '../../../utils/HelperFunctions'
import CustomTextInput3 from '../../Feeds/CommentsTextInput/CustomTextInput3'
import { useSelector } from 'react-redux'
import { translate } from '../../../i18n/TranslationService'

const EditMember = ({ projectId, onCancelAction, style }) => {
    const [email, setEmail] = useState('')
    const smallScreen = useSelector(state => state.smallScreen)
    const mobile = useSelector(state => state.smallScreenNavigation)
    const showGlobalSearchPopup = useSelector(state => state.showGlobalSearchPopup)
    const buttonItemStyle = { marginRight: smallScreen ? 8 : 4 }

    const inputTask = useRef()

    useEffect(() => {
        if (showGlobalSearchPopup) {
            onCancelAction()
        }
    }, [showGlobalSearchPopup])

    useEffect(() => {
        document.addEventListener('keydown', onKeyDown)
        return () => document.removeEventListener('keydown', onKeyDown)
    })

    const enterKeyAction = () => {
        if (email.trim().length > 0) {
            addProjectMember()
        }
    }

    const onKeyDown = ({ key }) => {
        if (key === 'Enter') {
            enterKeyAction()
        } else if (key === 'Escape') {
            onCancelAction()
        }
    }

    const addProjectMember = () => {
        const inviterUserId = store.getState().loggedUser.uid

        if (HelperFunctions.isValidEmail(email)) {
            Backend.inviteUserToProject(email, projectId, inviterUserId)
            onCancelAction()
            Keyboard.dismiss()
        }
    }

    return (
        <View style={[localStyles.container, mobile ? localStyles.containerUnderBreakpoint : undefined, style]}>
            <View style={[localStyles.inputContainer]}>
                <View style={[localStyles.icon, mobile ? localStyles.iconMobile : undefined]}>
                    <Icon name={'plus-square'} size={24} color={colors.Primary100} />
                </View>

                <CustomTextInput3
                    ref={inputTask}
                    placeholder={translate('Type an email to send invitation')}
                    containerStyle={[localStyles.input, mobile ? localStyles.inputUnderBreakpoint : undefined]}
                    autoFocus={true}
                    multiline={false}
                    numberOfLines={1}
                    onChangeText={text => setEmail(text)}
                    placeholderTextColor={colors.Text03}
                    disabledTags={true}
                    forceTriggerEnterActionForBreakLines={enterKeyAction}
                />
            </View>
            <View style={localStyles.buttonContainer}>
                <View style={[localStyles.buttonSection]}>
                    <View style={!mobile && { marginRight: 32 }} />
                </View>

                <View style={[localStyles.buttonSection, localStyles.buttonSectionRight]}>
                    {!smallScreen && (
                        <Button
                            title={translate('Cancel')}
                            type={'secondary'}
                            buttonStyle={buttonItemStyle}
                            onPress={onCancelAction}
                            shortcutText={'Esc'}
                        />
                    )}

                    <Button
                        title={smallScreen ? null : translate('Invite')}
                        type={'primary'}
                        icon={smallScreen ? 'plus' : null}
                        onPress={addProjectMember}
                        disabled={email.length === 0}
                        shortcutText={'Enter'}
                    />
                </View>
            </View>
        </View>
    )
}

export default EditMember

const localStyles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#ffffff',
        borderWidth: 1,
        borderStyle: 'solid',
        borderColor: colors.Grey200,
        borderRadius: 4,
        shadowColor: 'rgba(0,0,0,0.08)',
        shadowOffset: { width: 0, height: 4 },
        shadowOpacity: 1,
        shadowRadius: 8,
        elevation: 3,
        marginLeft: -16,
        marginRight: -16,
        marginBottom: 16,
    },
    containerUnderBreakpoint: {
        marginLeft: -8,
        marginRight: -8,
    },
    buttonContainer: {
        flex: 1,
        flexDirection: 'row',
        backgroundColor: colors.Grey100,
        borderTopWidth: 1,
        borderStyle: 'solid',
        borderTopColor: colors.Gray300,
        paddingVertical: 8,
        paddingHorizontal: 9,
    },
    buttonSection: {
        flexDirection: 'row',
        flexGrow: 1,
    },
    buttonSectionRight: {
        justifyContent: 'flex-end',
    },
    inputContainer: {
        minHeight: 56,
        flexDirection: 'row',
    },
    icon: {
        position: 'absolute',
        padding: 0,
        margin: 0,
        left: 23,
        top: 15,
    },
    iconMobile: {
        left: 15,
    },
    input: {
        ...styles.body1,
        paddingTop: 11,
        paddingBottom: 11,
        borderTopLeftRadius: 4,
        borderTopRightRadius: 4,
        paddingLeft: 67,
        paddingRight: 16,
        //  textAlignVertical: 'top',
    },
    inputUnderBreakpoint: {
        paddingLeft: 59,
        paddingRight: 8,
    },
})
