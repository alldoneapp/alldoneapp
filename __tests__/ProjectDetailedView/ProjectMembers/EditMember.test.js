/**
 * @jest-environment jsdom
 */

import React from 'react'
import EditMember from '../../../components/ProjectDetailedView/ProjectMembers/EditMember'
import renderer from 'react-test-renderer'

describe('EditMember component', () => {
    describe('EditMember snapshot test', () => {
        it('should render correctly', () => {
            const tree = renderer.create(<EditMember />).toJSON()
            expect(tree).toMatchSnapshot()
        })
    })
})
