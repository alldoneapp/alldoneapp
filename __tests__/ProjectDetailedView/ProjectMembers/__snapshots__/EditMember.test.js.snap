// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditMember component EditMember snapshot test should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#F1F3F4",
        "borderRadius": 4,
        "borderStyle": "solid",
        "borderWidth": 1,
        "elevation": 3,
        "flex": 1,
        "marginBottom": 16,
        "marginLeft": -16,
        "marginRight": -16,
        "shadowColor": "rgba(0,0,0,0.08)",
        "shadowOffset": Object {
          "height": 4,
          "width": 0,
        },
        "shadowOpacity": 1,
        "shadowRadius": 8,
      },
      undefined,
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "flexDirection": "row",
          "height": 56,
          "overflow": "hidden",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "left": 23,
            "margin": 0,
            "padding": 0,
            "position": "absolute",
            "top": 15,
          },
          undefined,
        ]
      }
    >
      <View>
        <Text
          style={
            Object {
              "color": "#007FFF",
              "fontFamily": "alldone",
              "fontSize": 24,
              "opacity": 1,
            }
          }
        >
          î§œ
        </Text>
      </View>
    </View>
    <TextInput
      allowFontScaling={true}
      autoFocus={true}
      multiline={false}
      numberOfLines={1}
      onChangeText={[Function]}
      onKeyPress={[Function]}
      placeholder="Type an email to send invitation"
      placeholderTextColor="#8A94A6"
      rejectResponderTermination={true}
      returnKeyType="done"
      style={
        Array [
          Object {
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "bottom": -1,
            "color": "#04142F",
            "flexWrap": "wrap",
            "fontFamily": "Roboto-Regular",
            "fontSize": 16,
            "left": -1,
            "letterSpacing": 0.32,
            "lineHeight": 24,
            "paddingBottom": 16,
            "paddingLeft": 68,
            "paddingRight": 16,
            "paddingTop": 16,
            "position": "absolute",
            "right": -1,
            "textAlignVertical": "top",
            "top": -1,
            "width": "101%",
          },
          undefined,
        ]
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#FAFBFB",
        "borderStyle": "solid",
        "borderTopColor": "#E7ECEF",
        "borderTopWidth": 1,
        "flex": 1,
        "flexDirection": "row",
        "paddingHorizontal": 9,
        "paddingVertical": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
          },
        ]
      }
    >
      <View
        style={
          Object {
            "marginRight": 32,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
            "flexGrow": 1,
          },
          Object {
            "justifyContent": "flex-end",
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#EAF0F5",
            "borderRadius": 4,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 40,
            "justifyContent": "center",
            "marginRight": 4,
            "maxHeight": 40,
            "minHeight": 40,
            "opacity": 1,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "flexWrap": "nowrap",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 14,
                  "letterSpacing": 0.8,
                  "lineHeight": 16,
                  "margin": 0,
                  "paddingHorizontal": 0,
                  "paddingVertical": 0,
                },
                Object {
                  "color": "#04142F",
                },
                Object {},
              ],
              Object {
                "marginLeft": 0,
              },
            ]
          }
        >
          Cancel
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "flex-start",
            "backgroundColor": "#0D55CF",
            "borderRadius": 4,
            "flexDirection": "row",
            "flexWrap": "nowrap",
            "height": 40,
            "justifyContent": "center",
            "maxHeight": 40,
            "minHeight": 40,
            "opacity": 1,
            "paddingLeft": 16,
            "paddingRight": 16,
            "paddingVertical": 8,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "#FFFFFF",
                  "flexWrap": "nowrap",
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 14,
                  "letterSpacing": 0.8,
                  "lineHeight": 16,
                  "margin": 0,
                  "paddingHorizontal": 0,
                  "paddingVertical": 0,
                },
                Object {
                  "color": "#ffffff",
                },
                Object {},
              ],
              Object {
                "marginLeft": 0,
              },
            ]
          }
        >
          Invite
        </Text>
      </View>
    </View>
  </View>
</View>
`;
